mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/subscribers/activity-modal.tsx

```diff
@@ -0,0 +1,33 @@
+import { Modal } from '@wordpress/components';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useEffect, useMemo, useState } from 'react';
+
+export function ActivityModal(): JSX.Element {
+  const history = useHistory();
+  const location = useLocation();
+  const pageParams = useMemo(
+    () => new URLSearchParams(location.search),
+    [location],
+  );
+  const [runId, setRunId] = useState(null);
+
+  // Open modal when activity param changes in the URL
+  useEffect(() => {
+    setRunId(pageParams.get('runId'));
+  }, [pageParams]);
```

Comment:
This was meant to prevent loading activity modal when any other URL parameter changes. This `useEffect` can be called many times when the URL changes, but the other one with calling an API will only be called if `runId` param changes. Do you know if there is a better way to handle this?

Reply:
I think you don't need the state for `runId`, you could probably just do:

```ts
const runId = pageParams.get('runId');
```

Then you can remove one state and one `useEffect`. Wouldn't that work the same? You'd be still passing only `runId` as a dependency to the data fetching effect. Would that work? (Didn't try it.)

Reply:
It did work, thanks! https://github.com/mailpoet/mailpoet/pull/5700/commits/8a2e8a59c6cdf0114482a15976d325b730f9934d